client/blocks/post-time/index.jsx

```diff
@@ -13,31 +13,40 @@ import Gridicon from 'gridicons';
  */
 import { getNormalizedPost } from 'state/posts/selectors';
 
-export function PostRelativeTime( { moment, post } ) {
-	let time;
+export function PostTime( { moment, post } ) {
+	let timeDisplay;
+
 	if ( post ) {
 		const { status, modified, date } = post;
-		time = includes( [ 'draft', 'pending' ], status ) ? modified : date;
+		const time = moment(
+			includes( [ 'draft', 'pending' ], status )
+				? modified
+				: date
+		);
+		if ( time.isBefore( moment().subtract( 7, 'days' ) ) ) {
+			// "August 30, 2017 4:46 PM" in English locale
+			timeDisplay = time.format( 'LLL' );
+		} else {
+			// "3 days ago" in English locale
+			timeDisplay = time.fromNow();
+		}
+	} else {
+		// Placeholder text: "a few seconds ago" in English locale
+		timeDisplay = moment().fromNow();
```

Comment:
If we set the default as the following, I believe we can remove this else block.
```
let timeDisplay = moment().fromNow();
```

Reply:
I wanted to keep all cases and calculations explicit here, so I moved this logic into a separate function in d826512873c70a9b6bdd53439e3f7e2d6fdcfba7.